widget/date_entry.go

```diff
@@ -0,0 +1,158 @@
+package widget
+
+import (
+	"time"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+)
+
+const dateFormat = "02 Jan 2006"
+
+// DateEntry is an input field which supports selecting from a fixed set of options.
+//
+// Since: 2.6
+type DateEntry struct {
+	Entry
+	Date      *time.Time
+	OnChanged func(*time.Time) `json:"-"`
+
+	dropDown *Calendar
+	popUp    *PopUp
+}
+
+// NewDateEntry creates a date input where the date can be selected or typed.
+//
+// Since: 2.6
+func NewDateEntry() *DateEntry {
+	e := &DateEntry{}
+	e.ExtendBaseWidget(e)
+	e.Wrapping = fyne.TextWrap(fyne.TextTruncateClip)
+	return e
+}
+
+// CreateRenderer returns a new renderer for this select entry.
+//
+// Implements: fyne.Widget
+func (e *DateEntry) CreateRenderer() fyne.WidgetRenderer {
+	e.ExtendBaseWidget(e)
+	e.Validator = func(in string) error {
+		_, err := time.Parse(dateFormat, in)
+		return err
+	}
+	e.Entry.OnChanged = func(in string) {
+		if in == "" {
+			e.Date = nil
+
+			if f := e.OnChanged; f != nil {
+				f(nil)
+			}
+		}
+		t, err := time.Parse(dateFormat, in)
+		if err != nil {
+			return
+		}
+
+		e.Date = &t
+
+		if f := e.OnChanged; f != nil {
+			f(&t)
+		}
+	}
+
+	if e.ActionItem == nil {
+		e.ActionItem = e.setupDropDown()
+		if e.Disabled() {
+			e.ActionItem.(fyne.Disableable).Disable()
+		}
+	}
+
+	return e.Entry.CreateRenderer()
+}
+
+// Enable this widget, updating any style or features appropriately.
+//
+// Implements: fyne.DisableableWidget
+func (e *DateEntry) Enable() {
+	if e.ActionItem != nil {
+		e.ActionItem.(fyne.Disableable).Enable()
```

Comment:
This is a public field, I'm not sure it's safe to always assume it is a fyne.Disableable if it is non-nil?